CleverFerret/src/main/java/com/universalmedialibrary/services/metadata/EnhancedMetadataService.kt

```
@@ -0,0 +1,330 @@
+            // Try different metadata keys for sample rate
+            retriever.extractMetadata(METADATA_KEY_SAMPLERATE)?.toIntOrNull()
+        } catch (e: Exception) {
+            null
+        }
+    }
+    
+    private fun getChannelCount(retriever: MediaMetadataRetriever): Int? {
+        return try {
+            // Try to extract channel count from available metadata
+            retriever.extractMetadata(METADATA_KEY_NUM_TRACKS)?.toIntOrNull()
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)